Packs/Tenable_sc/Integrations/Tenable_sc/Tenable_sc.py

```diff
@@ -53,116 +24,1033 @@ def get_server_url():
+
+        self.cookie = res.cookies.get('TNS_SESSIONID', self.cookie)
+        demisto.setIntegrationContext({'cookie': self.cookie})
+
+        return res.json()
+
+    def logout(self):
+        """
+        Send the request to logout for username & password authentication method.
+        """
+        self.send_request(path='token', method='delete')
+
+    def create_scan(self, args: Dict[str, Any]):
+        """
+        Send the request for create_scan_command and create_remediation_scan_command.
+        Args:
+            args (dict): The demisto.args() object.
+        Returns:
+            Dict: The response.
+        """
+        body = self.create_scan_body(args)
+
+        return self.send_request(path='scan', method='post', body=body)
+
+    def create_scan_body(self, args):
+        """
+        Construct the body for the create_scan request.
+        Args:
+            args (dict): The demisto.args() object.
+        Returns:
+            Dict: The request body.
+        """
+        create_scan_mapping_dict = {
+            'description': 'description',
+            'dhcpTracking': 'dhcp_tracking',
+            'timeoutAction': 'timeout_action',
+            'scanningVirtualHosts': 'scan_virtual_hosts',
+            'rolloverType': 'rollover_type',
+            'ipList': 'ip_list'
+        }
+        body = {key: args.get(value) for key, value in create_scan_mapping_dict.items() if args.get(value)}
+
+        scan_type = args.get("scan_type")
+        body["type"] = scan_type if scan_type else ('policy' if args.get("policy_id") else 'plugin')
+
+        body['name'] = args.get('name') or args.get('scan_name')
+
+        body["pluginID"] = args.get('plugin_id') or args.get('plugins_id')
+
+        if repo_id := args.get("repository_id"):
+            body["repository"] = {'id': repo_id}
+
+        if policy_id := args.get("policy_id"):
+            body["policy"] = {'id': policy_id}
+
+        if zone_id := args.get("zone_id"):
+            body["zone"] = {'id': zone_id}
+
+        if report_ids := args.get("report_ids"):
+            body['reports'] = [{'id': r_id, 'reportSource': 'individual'} for r_id in argToList(report_ids)]
+
+        if asset_ids := args.get("asset_ids"):
+            if str(asset_ids).startswith('All'):
+                manageable = True if asset_ids == 'AllManageable' else False
+                res = self.get_assets(None)
+                assets = get_elements(res['response'], manageable)
+                asset_ids = list(map(lambda a: a['id'], assets))
+            body['assets'] = [{'id': a_id} for a_id in argToList(asset_ids)]
+
+        if credentials := args.get("credentials"):
+            body['credentials'] = [{'id': c_id} for c_id in argToList(credentials)]
+
+        if max_scan_time := int(args.get('max_scan_time', '1')):
+            body['maxScanTime'] = max_scan_time * 3600
+
+        if schedule := args.get('schedule'):
+            body['schedule'] = {
```

Comment:
lets created an object here and only set it on body['schedule'] at the end so we dont need to keep referring to it by body['schedule']